packages/dds/tree/src/test/memory/tree.spec.ts

```diff
@@ -95,63 +154,151 @@ describe("SharedTree memory usage", () => {
 
 				public beforeIteration(): void {
-					this.sharedTree = createLocalSharedTree("testSharedTree");
+					this.sharedTree = createLocalSharedTree(
+						"testSharedTree",
+						RootNodeSchema,
+						new RootNodeSchema(initialState),
+					);
 				}
 			})(),
 		).timeout(40000); // Set relatively higher threshold as 100_000 iterations can take a while.
 
 		benchmarkMemory(
 			new (class implements IMemoryTestObject {
 				public readonly title =
-					`Set an optional integer property ${x} times in a local SharedTree, then clear it`;
+					`Set an optional integer property ${numberOfEntries} times in a local SharedTree, then clear it`;
 				private sharedTree?: TreeView<typeof RootNodeSchema>;
 
 				public async run(): Promise<void> {
 					assert(this.sharedTree?.root.child !== undefined);
 
-					for (let i = 0; i < x; i++) {
-						this.sharedTree.root.child.propertyOne = x;
+					for (let i = 0; i < numberOfEntries; i++) {
+						this.sharedTree.root.child.propertyOne = numberOfEntries;
 					}
 					this.sharedTree.root.child.propertyOne = undefined; // This is possible since the property is optional.
 				}
 
 				public beforeIteration(): void {
-					this.sharedTree = createLocalSharedTree("testSharedTree");
+					this.sharedTree = createLocalSharedTree(
+						"testSharedTree",
+						RootNodeSchema,
+						new RootNodeSchema(initialState),
+					);
 				}
 			})(),
 		).timeout(40000); // Set relatively higher threshold as 100_000 iterations can take a while.
 	}
+
+	function runBenchmarkMemoryForSubTree<TSchema extends ImplicitFieldSchema>(
+		title: string,
+		schema: TSchema,
+		generateContent: (numberOfNodes: number) => InsertableTreeFieldFromImplicitField<TSchema>,
+		testNodeCounts: number[],
+	) {
+		describe(title, () => {
+			for (const numberOfNodes of testNodeCounts) {
+				for (const forestType of [ForestType.Reference, ForestType.Optimized]) {
+					benchmarkMemory(
+						new (class implements IMemoryTestObject {
+							public readonly title =
+								`initialize ${numberOfNodes} nodes into tree using ${forestType === 0 ? "ObjectForest" : "ChunkedForest"}`;
+
+							private sharedTree: TreeView<typeof schema> | undefined;
+
+							public async run(): Promise<void> {
+								this.sharedTree = createLocalSharedTree(
+									"testSharedTree",
+									schema,
+									generateContent(numberOfNodes),
+									{
+										forest: forestType,
+										treeEncodeType: TreeCompressionStrategy.Compressed,
+									},
+								);
+							}
+
+							public beforeIteration(): void {
+								this.sharedTree = undefined;
+							}
+						})(),
+					).timeout(400000);
+				}
+			}
+		});
+	}
+
+	const numberOfNodesForTests = isInPerformanceTestingMode ? [1, 10, 100, 1000] : [10];
+	// TODO: needs .only to describe block when running in performance mode if you need to see the results. Check to see why it does not run without .only.
+	describe("Forest memory usage", () => {
+		runBenchmarkMemoryForSubTree(
```

Comment:
I don't think "run" is the best verb to use here, since this method does not actually run the benchmarks. It only enumerates them. They aren't actually run until mocha executes the tests. Mocha uses the verb "describe" when defining a suite of tests, and I think aligning with that would be good. Also "subtree" is generally one word no need for a capital T in it so `describeMemoryBenchmarksForSubtrees`